tests/test_20_use_case_topology.py

```diff
@@ -229,5 +229,106 @@ def test_011_intra_domain_link_down_path_found(self):
         assert len(initial_topology["nodes"]) == len(updated_topology["nodes"]), "Number of nodes changed"
         assert len(initial_topology["links"]) == len(updated_topology["links"]), "Number of links changed"
 
+    @pytest.mark.xfail(reason="The L2VPN status remains up after changing the status of an associated node to down")
+    def test_040_node_down(self):
+        """
+        Test Use Case 4: Node Status Change (Down)
+        
+        Verify that when a node goes down:
+        1. SDX Controller updates the status of the node, its ports, and links
+        2. SDX Controller updates affected L2VPN statuses to "error"
+        3. SDX Controller refuses to provision new services on the node
+        """
+
+        # Create a L2VPN that will use the node
+        l2vpn_api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+        l2vpn_payload = {
+            "name": "Test L2VPN for node down",
+            "endpoints": [
+                {
+                    "port_id": "urn:sdx:port:ampath.net:Ampath1:50",
+                    "vlan": "300",
+                },
+                {
+                    "port_id": "urn:sdx:port:tenet.ac.za:Tenet01:50",
+                    "vlan": "300",
+                }
+            ]
+        }
+        response = requests.post(l2vpn_api_url, json=l2vpn_payload)
+        assert response.status_code == 201, response.text
+        l2vpn_id = response.json().get("service_id")
+
+        # Wait for L2VPN to be provisioned
+        time.sleep(5)
+
+        response = requests.get(l2vpn_api_url)
+        assert response.status_code == 200, response.text
+        l2vpn_status = response.json().get(l2vpn_id).get("status")
+        assert l2vpn_status == "up", f"L2VPN status should be up, but is {l2vpn_status}"
+
+        h1, h6 = self.net.net.get('h1', 'h6')
+        h1.cmd('ip link add link %s name vlan300 type vlan id 300' % (h1.intfNames()[0]))
+        h1.cmd('ip link set up vlan300')
+        h1.cmd('ip addr add 10.3.1.1/24 dev vlan300')
+        h6.cmd('ip link add link %s name vlan300 type vlan id 300' % (h6.intfNames()[0]))
+        h6.cmd('ip link set up vlan300')
+        h6.cmd('ip addr add 10.3.1.6/24 dev vlan300')
+
+        # test connectivity
+        assert ', 0% packet loss,' in h1.cmd('ping -c4 10.3.1.6')
+
+        node = self.net.net.get('Ampath1')
+        
+        config = self.net.change_node_status('down')
+
+        time.sleep(15)
+        
+        # status of the node should be down
+        api_url_topo = SDX_CONTROLLER + '/topology'
+        response_topology = requests.get(api_url_topo)
+
+        response_l2vpn = requests.get(l2vpn_api_url)
+        l2vpn_status = response_l2vpn.json().get(l2vpn_id).get("status")
```

Comment:
Suggestion: instead of just saving the status, let's save all data and then we just filter when doing the assert later on

```suggestion
        l2vpn_status = response_l2vpn.json()
```